### PR TITLE
MINOR: [Java] Update installation documentation for BOM file

### PR DESCRIPTION
**Pull request by jduo**
Update the installation documentation to describe how to use the bill of materials module.

### Rationale for this change
Add information to get users to use the bill of materials file.

### What changes are included in this PR?
Provide an example of including the bill of materials dependency.

### Are these changes tested?
N/A

### Are there any user-facing changes?
Documentation

### CODE REVIEW

**Comment by jduo** [comment]
@lidavidm @danepitkin

File: docs/source/java/install.rst [hunk 1]
```diff
@@ -83,6 +83,40 @@ arrow-vector, and arrow-memory-netty.
         </dependencies>
     </project>
 
+A bill of materials (BOM) module has been provided to simplify adding
+Arrow modules. This eliminates the needed to specify the version for
```
Comment (danepitkin) [inline docs/source/java/install.rst:87]:
```suggestion
Arrow modules. This eliminates the need to specify the version for
```

**Review by danepitkin — COMMENTED** [review]
One small suggestion, otherwise LGTM!

File: docs/source/java/install.rst [hunk 2]
```diff
@@ -83,6 +83,40 @@ arrow-vector, and arrow-memory-netty.
...
+    <?xml version="1.0" encoding="UTF-8"?>
+    <project xmlns="http://maven.apache.org/POM/4.0.0"
+             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+             xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+        <modelVersion>4.0.0</modelVersion>
+        <groupId>org.example</groupId>
+        <artifactId>demo</artifactId>
+        <version>1.0-SNAPSHOT</version>
+        <properties>
+            <arrow.version>9.0.0</arrow.version>
```
Comment (lidavidm) [inline docs/source/java/install.rst:101]:
```suggestion
            <arrow.version>15.0.0</arrow.version>
```

**Review by lidavidm — APPROVED** [review]
Thanks, just one nit

File: docs/source/java/install.rst (hunk 2 above)
Comment (danepitkin) [reply docs/source/java/install.rst]: Might as well change the above example, too!

**Pull request merged**